app/assets/javascripts/labels.coffee

```diff
@@ -1,35 +1,82 @@
 ready = ->
   $('.my-tag').select2
+    width: '100%'
     tags: true
     multiple: true
     minimumInputLength: 1
+    maximumSelectionLength: 5
+    closeOnSelect: false
+
+    ajax: {
+      url: '/labels/search',
+      dataType: 'json',
+      data: (params) ->
+        query = {
+          search: params.term,
+          page: params.page || 1
+        }
+        return query;
+    }
+
+    ajax: {
+      url: '/labels/search',
+      dataType: 'json',
+      data: (params) ->
+        query = {
+          search: params.term,
+          page: params.page || 1
+        }
+        return query;
+    }
```

Comment:
I struggled a bit with git because I had to rebase this branch onto the newer reimplement_labels branch and there were some conflicts. That is probably where this comes from.